src/agent/README.md

```diff
@@ -1,79 +1,100 @@
-| exec/list process       | :white_check_mark: |
-| I/O stream              | :white_check_mark: |
-| Cgroups                 | :white_check_mark: |
-| Capabilities, `rlimit`, readonly path, masked path, users | :white_check_mark: |
-| Seccomp                 | :white_check_mark: |
-| container stats (`stats_container`)                     | :white_check_mark: |
-| Hooks                   | :white_check_mark: |
-| **Agent Features & APIs** |
-| run agent as `init` (mount fs, udev, setup `lo`) | :white_check_mark: |
-| block device as root device                      | :white_check_mark: |
-| Health API                                       | :white_check_mark: |
-| network, interface/routes (`update_container`)   | :white_check_mark: |
-| File transfer API (`copy_file`)                  | :white_check_mark: |
-| Device APIs (`reseed_random_device`, , `online_cpu_memory`, `mem_hotplug_probe`, `set_guet_data_time`) | :white_check_mark: |
-| VSOCK support                                    | :white_check_mark: |
-| virtio-serial support                            | :heavy_multiplication_x: |
-| OCI Spec validator                               | :white_check_mark: |
-| **Infrastructures**|
-| Debug Console | :white_check_mark: |
-| Command line  | :white_check_mark: |
-| Tracing       | :heavy_multiplication_x: |
-
-## Getting Started
-
-### Build from Source
-The rust-agent needs to be built statically and linked with `musl`
-
-> **Note:** skip this step for ppc64le, the build scripts explicitly use gnu for ppc64le.
+# Kata Agent
+
+## Overview
+
+The Kata agent is a long running process that runs inside the Virtual Machine
+(VM) (also known as the "pod" or "sandbox").
+
+The agent is packaged inside the Kata Containers
+[guest image](../../docs/design/architecture.md#guest-image)
+which is used to boot the VM. Once the runtime has launched the configured
+[hypervisor](../../docs/hypervisors.md) to create a new VM, the agent is
+started. From this point on, the agent is responsible for creating and
+managing the life cycle of the containers inside the VM.
+
+The [runtime](../runtime) sends [API commands](protocols/protos) to the agent
+using the `ttRPC` protocol.
+
+For further details, see the [architecture
+document](../../docs/design/architecture.md).
+
+## Audience
+
+If you simply wish to use Kata Containers, it is not necessary to understand
+the details of how the agent operates. Please see the
+[installation documentation](../../docs/install) for details of how to install
+a Kata Containers system (which will include the Kata agent).
+
+The remainder of this document is only useful for developers and testers.
+
+## Build from Source
+
+The agent is written in the rust language and needs to be built statically and
+linked with the `musl` C library.
+
+> **Note:**
+>
+> Skip this step for ppc64le: the build scripts explicitly use the GNU C
+> library on this platform.
+
+### Platform specific instructions
+
+#### All platforms except PPC64le
```

Comment:
... and s390x?

Reply:
s390x also needs protoc at buildtime, yes @jodh-intel

Reply:
@Jakob-Naucke - ok, updated.

Reply:
I think this is incorrect: you shouldn't need the `protoc` command unless you're modifying the `.proto` files, which doesn't happen in an agent build. I'll rework to add a new section about modifying the API...

Reply:
I have never dug into why, but a standard agent build fails on s390x without having protoc.

Reply:
```
error: failed to run custom build command for `prost-build v0.5.0`

Caused by:
  process didn't exit successfully: `/home/ubuntu/go/src/github.com/kata-containers/kata-containers/src/agent/target/release/build/prost-build-d01ef8d4f6fe2b4f/build-script-build` (exit status: 101)
  --- stderr
  thread 'main' panicked at 'Failed to find the protoc binary. The PROTOC environment variable is not set, there is no bundled protoc for this platform, and protoc is not in the PATH', /home/ubuntu/.cargo/registry/src/github.com-eae4ba8cbf2ce1c7/prost-build-0.5.0/build.rs:97:10
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: build failed
make: *** [Makefile:110: target/s390x-unknown-linux-gnu/release/kata-agent] Error 101
```

Reply:
@liubin - Do you know why `protoc` is required here? `prost` only appears in `Cargo.lock` and we already have the golang output of `protoc` stored in git to avoid the need to run it... atleast that had been my understanding.